extensions/amp-analytics/0.1/variables.js

```diff
@@ -126,78 +109,39 @@ export class VariableService {
     /** @private {!Window} */
     this.win_ = window;
 
-    /** @private {!Object<string, !Filter>} */
-    this.filters_ = map();
-
-    this.register_('default', new Filter(defaultFilter, /* allowNulls */ true));
-    this.register_('substr', new Filter(substrFilter));
-    this.register_('trim', new Filter(value => value.trim()));
-    this.register_('json', new Filter(value => JSON.stringify(value)));
-    this.register_('toLowerCase', new Filter(value => value.toLowerCase()));
-    this.register_('toUpperCase', new Filter(value => value.toUpperCase()));
-    this.register_('not', new Filter(value => String(!value)));
-    this.register_('base64', new Filter(value => btoa(value)));
-    this.register_('hash', new Filter(this.hashFilter_.bind(this)));
-    this.register_('if', new Filter(
-        (value, thenValue, elseValue) => value ? thenValue : elseValue, true));
+    /** @private {!Object<string, *>} */
+    this.filters_ = {};
+
+    this.register_('DEFAULT', defaultFilter);
+    this.register_('SUBSTR', substrFilter);
+    this.register_('TRIM', value => value.trim());
+    this.register_('JSON', value => JSON.stringify(value));
+    this.register_('TOLOWERCASE', value => value.toLowerCase());
+    this.register_('TOUPPERCASE', value => value.toUpperCase());
+    this.register_('NOT', value => String(!value));
+    this.register_('BASE64', value => btoa(value));
+    this.register_('HASH', this.hashFilter_.bind(this));
+    this.register_('IF',
+        (value, thenValue, elseValue) => value ? thenValue : elseValue);
+  }
+
+  /**
+   * @return {!Object} contains all registered filters
+   */
+  getFilters() {
+    return this.filters_;
   }
 
   /**
    * @param {string} name
-   * @param {!Filter} filter
+   * @param {*} filter
```

Comment:
`{function(*)}` ?

Reply:
I think the confusion here is that filter is a bad name. In reality these are macros that can be pretty much anything: primitives, functions, promises, etc